service/src/main/java/uk/gov/hmcts/reform/fpl/service/validators/GroundsValidator.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.hmcts.reform.fpl.service.validators;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.validation.groups.EPOGroup;
+
+import java.util.List;
+import javax.validation.groups.Default;
+
+@Component
+public class GroundsValidator extends PropertiesValidator {
+
+    @Override
+    public List<String> validate(CaseData caseData) {
+        if (hasEmergencyProtectionOrder(caseData)) {
+            return super.validate(caseData, List.of("grounds", "groundsForEPO"), Default.class, EPOGroup.class);
+        } else {
+            return super.validate(caseData, List.of("grounds"));
+        }
+    }
+
+    private boolean hasEmergencyProtectionOrder(CaseData caseData) {
```

Comment:
Do you think this would be more useful in case data or not? It sort of fits within the domain scope of CaseData.

Reply:
for no it is needed only here i guess, CaseData already is too big

Reply:
It really is